apps/condo/domains/organization/schema/FindOrganizationsByTinService.js

```diff
@@ -0,0 +1,136 @@
+    checkTotalRequestLimitCountersByUser,
+} = require('@condo/domains/user/utils/serverSchema/requestLimitHelpers')
+
+
+const appLogger = getLogger('condo')
+const logger = appLogger.child({ module: 'organization/findOrganizationsByTin' })
+
+const MAX_TOTAL_REQUESTS = 50
+const UNAVAILABLE_TINS = ['0000000000', '000000000000']
+
+/**
+ * List of possible errors, that this custom schema can throw
+ * They will be rendered in documentation section in GraphiQL for this custom schema
+ */
+const ERRORS = {
+    EMPTY_TIN: {
+        query: 'findOrganizationsByTin',
+        variable: ['data', 'tin'],
+        code: BAD_USER_INPUT,
+        type: 'EMPTY_TIN',
+        message: 'Empty tin',
+    },
+    UNAVAILABLE_TIN: {
+        query: 'findOrganizationsByTin',
+        variable: ['data', 'tin'],
+        code: BAD_USER_INPUT,
+        type: 'UNAVAILABLE_TIN',
+        message: 'Unavailable tin',
+    },
+    DV_VERSION_MISMATCH: {
+        ...COMMON_ERRORS.DV_VERSION_MISMATCH,
+        query: 'findOrganizationsByTin',
+    },
+    WRONG_SENDER_FORMAT: {
+        ...COMMON_ERRORS.WRONG_SENDER_FORMAT,
+        query: 'findOrganizationsByTin',
+    },
+}
+
+const FindOrganizationsByTinService = new GQLCustomSchema('FindOrganizationsByTinService', {
+    types: [
+        {
+            access: true,
+            type: 'input FindOrganizationsByTinInput { dv: Int!, sender: JSON!, tin: String! }',
+        },
+        {
+            access: true,
+            type: 'type FindOrganizationsByTinOrganizationType { id: String!, name: String! }',
+        },
+        {
+            access: true,
+            type: 'type FindOrganizationsByTinOutput { organizations: [FindOrganizationsByTinOrganizationType]! }',
+        },
+    ],
+
+    queries: [
+        {
+            access: access.canFindOrganizationsByTin,
+            schema: 'findOrganizationsByTin (data: FindOrganizationsByTinInput!): FindOrganizationsByTinOutput',
+            doc: {
+                summary: 'Returns all organizations by TIN. Used to get organizations that the user can apply to join',
+                errors: ERRORS,
+            },
+            resolver: async (parent, args, context) => {
+                const { data } = args
+                const { tin, dv, sender } = data
+                const authedItemId = get(context, 'authedItem.id', null)
+                if (!authedItemId) throw new Error('no authedItemId!')
+
+                checkDvAndSender(data, ERRORS.DV_VERSION_MISMATCH, ERRORS.WRONG_SENDER_FORMAT, context)
+
+                if (!tin) throw new GQLError(ERRORS.EMPTY_TIN, context)
+
+                // NOTE: we don't use "isFeatureEnabled" because it can skip the request limit for all users
+                const userWhiteList = await featureToggleManager.getFeatureValue(context, USER_WHITE_LIST_FOR_FIND_ORGANIZATIONS_BY_TIN, [], { user: authedItemId }) || []
+                const skipRequestLimit = get(context, 'authedItem.isAdmin', false)
```

Comment:
I do not really like that we store some access-related logic inside of the mutation.. 

I'd have moved it to access and exported it as skipRateLimit() function

This way if I had to change something related to access checks (for example add a special permission that allows you to bypass the rate limit) i have to change just one file

Again, just a nitpick

Reply:
I have not seen such logic in our accesses